src/raftstore/store/peer.rs

```diff
@@ -404,6 +415,40 @@ impl Peer {
         down_peers
     }
 
+    pub fn check_stale_state(&mut self, d: Duration) -> StaleState {
+        // update leader_missing_time according to current state
+        if self.leader_id() == raft::INVALID_ID {
+            if self.leader_missing_time.is_none() {
+                self.leader_missing_time = Some(Instant::now())
+            }
+        } else if self.is_initialized() {
+            // When a peer is not initialized, it has no data at storage.
```

Comment:
A peer is considered as in the `leader missing` state if it's uninitialized or if it is initialized but is isolated from its leader.
For an uninitialized peer, even if its leader sends heartbeats to it, it cannot successfully receive the snapshot from the leader and apply the snapshot. The Raft state machine cannot work in an uninitialized peer to detect if the leader is working. For a peer that is initialized but is isolated from its leader, there is no way to detect the state of the leader.